src/apps/js_v8/js_v8_base.cpp

```diff
@@ -0,0 +1,529 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#include "apps/utils/metrics_tracker.h"
+#include "ccf/app_interface.h"
+#include "ccf/historical_queries_adapter.h"
+#include "ccf/user_frontend.h"
+#include "ccf/version.h"
+#include "crypto/entropy.h"
+#include "crypto/key_wrap.h"
+#include "crypto/rsa_key_pair.h"
+#include "kv/untyped_map.h"
+#include "kv_module_loader.h"
+#include "named_auth_policies.h"
+#include "tmpl/ccf_global.h"
+#include "tmpl/console_global.h"
+#include "tmpl/request.h"
+#include "v8_runner.h"
+#include "v8_util.h"
+
+#include <memory>
+#include <stdexcept>
+#include <vector>
+
+using namespace std;
+using namespace ccf;
+using namespace kv;
+
+namespace ccfapp
+{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wc99-extensions"
+
+  /**
+   * V8 Handlers, holds the list of handlers from a JavaScript source to be
+   * called via RPC (through RPCFrontend).
+   */
+  class V8Handlers : public UserEndpointRegistry
+  {
+    NetworkTables& network;
+    ccfapp::AbstractNodeContext& node_context;
+    ::metrics::Tracker metrics_tracker;
+
+    void execute_request(
+      const ccf::endpoints::EndpointProperties& props,
+      ccf::endpoints::EndpointContext& endpoint_ctx)
+    {
+      if (props.mode == ccf::endpoints::Mode::Historical)
+      {
+        // Historical mode need a v2 adapter (why?)
```

Comment:
The v2 adapter has marginally more precise http return codes (https://github.com/microsoft/CCF/blob/623f403b817b4364cc2a98931642a951ff5b8247/CHANGELOG.md#added-3), in response to https://github.com/microsoft/CCF/issues/2640.

Reply:
I removed the comment, left-over from js_generic.